tests/hypothesis_strategies/constraints.py

```diff
@@ -0,0 +1,149 @@
+"""Hypothesis strategies for constraints."""
+
+from functools import partial
+from typing import Any, List, Optional, Type, Union
+
+import hypothesis.strategies as st
+
+from baybe.constraints.conditions import (
+    SubSelectionCondition,
+    ThresholdCondition,
+    _valid_logic_combiners,
+)
+from baybe.constraints.continuous import (
+    ContinuousLinearEqualityConstraint,
+    ContinuousLinearInequalityConstraint,
+)
+from baybe.constraints.discrete import (
+    DiscreteExcludeConstraint,
+    DiscreteLinkedParametersConstraint,
+    DiscreteNoLabelDuplicatesConstraint,
+    DiscreteProductConstraint,
+    DiscreteSumConstraint,
+)
+from baybe.parameters.base import DiscreteParameter
+from baybe.parameters.numerical import NumericalDiscreteParameter
+
+
+def sub_selection_conditions(superset: Optional[List[Any]] = None):
```

Comment:
didnt you introduce hooks that avoid `List` and have `list` everywhere? Hook not installed? This branch not rebased?

Reply:
Yes correct, but not yet active in this branch. Once rebased, we'll get the error (tested it). So will close here